extension/parquet/parquet_extension.cpp

```diff
@@ -45,8 +45,14 @@
 namespace duckdb {
 
 struct ParquetReadBindData : public TableFunctionData {
+	// TODO: the file list mutex is currently required to do thread-safe access into the file_list in the bind_data
+	unique_ptr<mutex> file_list_lock;
```

Comment:
Can you maybe turn these into private variables and have methods to access them?

Reply:
Going to lock the MultiFileList internally instead